proxy/vm.go

```diff
@@ -150,9 +150,10 @@ func (vm *vm) ioHyperToClients() {
 
 		err = hyperstart.SendIoMessageWithConn(session.client, msg)
 		if err != nil {
-			fmt.Fprintf(os.Stderr,
-				"error writing I/O data to client: %v\n", err)
-			break
+			// When the shim is forcefully killed, it's possible we
+			// still have data to write. Ignore errors for that case.
+			vm.infof(1, "io", "error writing I/O data to client:", err)
+			continue
```

Comment:
vm.hyperHandler.ReadIoMessage() is a blocking read() on the hyperstart socket. It will error out with EOF when the qemu process dies.

If the qemu process doesn't go away, this goroutine doesn't either, but that's the intended behaviour: at the moment the proxy is only slave to the runtime/qemu and reacts to command/events from those.